dialogy/plugins/text/address_parser/maps.py

```diff
@@ -0,0 +1,150 @@
+import os
+import re
+import uuid
+from typing import Type, List, Dict, Any, Optional, Union
+
+import googlemaps
+
+from dotenv import load_dotenv
+
+from loguru import logger
+from dialogy.plugins.text.address_parser import mapmyindia
+
+load_dotenv(override=True)
+GOOGLE_MAPS_API_TOKEN = os.getenv("GOOGLE_MAPS_API_TOKEN")
+gmaps = googlemaps.Client(key=GOOGLE_MAPS_API_TOKEN)
```

Comment:
Lets expect credentials as input within the plugin for both google and mapmyindia. 

The function `get_matching_address_from_gmaps_autocomplete` can use a fully initialized client. (re-initialization could cause a network overhead).

I would rephrase myself like so:

```python

        provider_functions: Dict[str, Callable[..., str]] = {
            "google": get_matching_address_from_gmaps_autocomplete(gmaps),
            "mmi": get_address_from_mapmyindia_geocode(mmi),
        }
```
and define the provider functions as closures to hold the client.

```python

def get_matching_address_from_gmaps_autocomplete(client: Type):
    def wrapper(
        most_confident_transcription: str,
        country_code: str = "in",
        language: str = "en",
        pin: str = "",
        **kwargs: Any,
    ) -> str:
        # The logic written within `get_matching_address_from_gmaps_autocomplete`
        ...
        return matching_address
   return wrapper
```

Reply:
I would also avoid `**kwargs` as they are not used.